src/app/components/app-settings/app-settings.component.ts

```diff
@@ -37,10 +37,20 @@ export class AppSettingsComponent implements OnInit {
       name: null,
       email: null
     };
+
+    this.auth.loginStatus.subscribe(loggedIn => {
```

Comment:
this cannot work, `loginStatus` is fired on login / logout event.. I believe you want to call `this.auth.getUserDetails` directly, without that.